src/server/server_family.cc

```diff
@@ -549,6 +550,17 @@ void ServerFamily::Init(util::AcceptServer* acceptor, std::vector<facade::Listen
       LOG(WARNING) << "Invalid snapshot time specifier " << save_time;
     }
   }
+
+  string snapshot_cron_exp = GetFlag(FLAGS_snapshot_cron);
+  if (!snapshot_cron_exp.empty()) {
+    try {
+      auto cron = cron::make_cron(snapshot_cron_exp);
+      snapshot_schedule_fb_ = service_.proactor_pool().GetNextProactor()->LaunchFiber(
+          [cron, this] { SnapshotScheduling(cron); });
+    } catch (cron::bad_cronexpr const& ex) {
+      LOG(WARNING) << "Invalid cron expression: " << ex.what();
+    }
+  }
```

Comment:
I would wrap this in a function and call it from here.
To clarify, I'd also move the logic for `save_snapshot` into that function.
While there, and given we only want 1 schedule, I'd try to consolidate the logic somehow (maybe only use cron, and convert schedule to cron if it set?)

Reply:
Sounds interesting.
`save_schedule` flag as defined the old way saves a snapshot every 24h at that time? Or is that the interval in which a snapshot is saved? The former will probably be easier to implement I think.

Reply:
From reading the code (I've never actually used it), it looks like the format is HH:MM, where HH/MM can be numbers or '*' (or both, like 1*).
There's quite a lot of supporting code for that, and I _think_ (although I may be missing something) that it could be pretty easily migrated to a cron-like string(?)

Reply:
Yup, I figured a way to make a cron expression from that just like you suggested. However it seems that there's a drawback of potentially skipping the first save, and the actual first save will be after more than 24h...
I'm still thinking of an elegant solution for this.

Reply:
Can you kindly explain why the skip might happen? Will it only happen upon conversion, and not for `--snapshot_cron` flag?

Reply:
Yes, it might be just me misunderstanding how to construct the correct cron expression😅
I will implement it correctly (I hope) soon.